File: src/main/java/dev/mayaqq/estrogen/integrations/ModMenuIntegration.java
package dev.mayaqq.estrogen.integrations;

import com.terraformersmc.modmenu.api.ConfigScreenFactory;
import com.terraformersmc.modmenu.api.ModMenuApi;
import dev.mayaqq.estrogen.config.EstrogenConfig;

public class ModMenuIntegration implements ModMenuApi {
    @Override
    public ConfigScreenFactory<?> getModConfigScreenFactory() {
        return EstrogenConfig.INSTANCE::makeScreen;
    }
}

File: src/fabric/kotlin/dev/mayaqq/estrogen/fabric/compat/ModMenuIntegration.kt
package dev.mayaqq.estrogen.fabric.compat

import com.terraformersmc.modmenu.api.ConfigScreenFactory
import com.terraformersmc.modmenu.api.ModMenuApi
import dev.mayaqq.estrogen.client.content.screen.EstrogenMenuScreen

class ModMenuIntegration : ModMenuApi {
    override fun getModConfigScreenFactory(): ConfigScreenFactory<*> {
        return ConfigScreenFactory { screen -> EstrogenMenuScreen(screen) }
    }
}